### ISSUE TITLE
Invalid offset name for timezone without full hour offset

### ISSUE THREAD

**Issue by omatrycy** [briannesbitt/Carbon#2059]
Hello,

I encountered an issue with the following code:
```php
\Carbon\CarbonTimeZone::create('Pacific/Marquesas')->toOffsetName()
```

Carbon version: 2.31.0

PHP version: 7.2.25

I expected to get:

```
-09:30
```

But I actually get:

```
-10:30
```

Timezone Pacific/Marquesas has offset -09:30, no summer time (please to check https://www.zeitverschiebung.net/en/timezone/pacific--marquesas).

The bug is in line [code](https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/CarbonTimeZone.php#L157)
```
$hours = floor($minutes / 60);
```
It floors 9.5 to 10. It should get integer value of the number. Minutes are calculated below. This bug is probably for all timezones without full hour (like : +13:45)

Thanks!

**Comment by kylekatarnls** [comment]
Indeed floor truncate positive values, but it shift down negative values. So it's for all timezones with not-zero minutes and < GMT.

**Comment by kylekatarnls** [comment]
It sounds like using `intval` instead of `floor` would fix it as intval(-9.5) is -9. Also having a unit test to check more different timezones would be great.

**Comment by kylekatarnls** [comment]
This will be released in 2.33, but you can right now test your fix using `"nesbot/carbon": "dev-master"` in **composer.json**.
